73466

SAM19991 opened the pull request.

SAM19991 committed:
73466

SAM19991 committed:
73466

SAM19991 committed:
73466

MarselYsup commented on src/main/java/school/faang/bjs273466/Boss.java:
```diff
@@ -0,0 +1,39 @@
+    private final int maxPlayers;
+    private final List<Player> currentPlayers = new ArrayList<>();
+
+    public Boss(int maxPlayers) {
```
используй ```@RequiredArgsConstructor``` это будет эквивалентно, однако такой способ будет более гибким - так как ломбок автоматически генерирует конструктор

MarselYsup commented on src/main/java/school/faang/bjs273466/Boss.java:
```diff
@@ -0,0 +1,39 @@
+                            player.getName(), Thread.currentThread().getName());
+                    this.wait();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
```
тут нужно добавить 
```java
Thread.currentThread().interrupt()
```
 , чтобы установить флаг прерывания текущего потока. Это позволяет потоку узнать, что его просят завершить работу или прервать текущую операцию.

MarselYsup commented on src/main/java/school/faang/bjs273466/Player.java:
```diff
@@ -0,0 +1,20 @@
+    public void doBattle(Boss boss) {
+        boss.joinBattle(this);
+        try {
+            Thread.sleep(2000);
```
Магическое число - лучше вынеси в константу

MarselYsup commented on src/main/java/school/faang/bjs273466/Player.java:
```diff
@@ -0,0 +1,20 @@
+        } catch (InterruptedException e) {
+            System.out.println("Battle interrupted");
+        }
+        boss.leaveBattle(this);
```
можно было бы вынести в ```finally``` блок, если бы ты пробрасывал ошибку в catch дальше

MarselYsup commented on src/main/java/school/faang/bjs273466/Main.java:
```diff
@@ -0,0 +1,33 @@
+    public static void main(String[] args) {
+        Boss boss = new Boss(5);
+        List<Player> players = new ArrayList<>();
+        for (int i = 1; i < 15; i++) {
```
можно в функциональном стиле 

```java
        IntStream.range(1, 15).forEach(index -> players.add(new Player("Player %d".formatted(index))));

```

SAM19991 replied:
Вот это круто, я о таком не знал

MarselYsup commented on src/main/java/school/faang/bjs273466/Main.java:
```diff
@@ -0,0 +1,33 @@
+    private static final int WAIT_TIME = 1;
+
+    public static void main(String[] args) {
+        Boss boss = new Boss(5);
```
можно вынести в константу для гибкости

MarselYsup commented on src/main/java/school/faang/bjs273466/Player.java:
```diff
@@ -0,0 +1,20 @@
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            System.out.println("Battle interrupted");
```
тут нужно добавить 
```java
Thread.currentThread().interrupt()
```
 , чтобы установить флаг прерывания текущего потока. Это позволяет потоку узнать, что его просят завершить работу или прервать текущую операцию.

SAM19991 committed:
Использовал @RequiredArgsConstructor вместо стандартного конструктора, для гибкости.
Добавил Thread.currentThread().interrupt() в catch блок метода joinBattle.
Вынес в константу максимальное количество игроков.
Переписал цикл for в функциональном стиле.

SAM19991 committed:
Использовал @RequiredArgsConstructor вместо стандартного конструктора, для гибкости.
Добавил Thread.currentThread().interrupt() в catch блок метода joinBattle.
Вынес в константу максимальное количество игроков.
Переписал цикл for в функциональном стиле.
